Maths/IsLeapYear.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * @function IsLeapYear
+ * @description Checks if a year is a leap year (Gregorian calendar).
+ * A year is a leap year if it is divisible by 4 but not by 400 or if it is divisible by 400.
+ * @param {number} year - A year, natural number > 0.
+ * @return {boolean} - True if given year is a leap year.
+ * @see https://en.wikipedia.org/wiki/Leap_year#Gregorian_calendar
+ * @example IsLeapYear(2000) = true
+ * @example IsLeapYear(2001) = false
+ */
+
+export const IsLeapYear = (year: number): boolean => {
+  if (year <= 0 || !Number.isInteger(year)) {
+    throw new Error("year must be a natural number > 0");
+  }
+
+  if (year % 4 > 0) {
+    return false;
+  }
+
+  if (year % 100 > 0) {
+    return true;
+  }
+
+  if (year % 400 > 0) {
+    return false;
+  }
+
+  return true;
```

Comment:
> Every year that is exactly divisible by four is a leap year, except for years that are exactly divisible by 100, but these centurial years are leap years if they are exactly divisible by 400.

You can compress all your `if`s to `return year % 4 == 0 && (year % 100 != 0 || year % 400 == 0);`

Reply:
I assumed that better readability is preferred, but I implemented your suggestion.